cpp/src/arrow/type.cc

```diff
@@ -219,6 +219,15 @@ std::string UnionType::ToString() const {
 // ----------------------------------------------------------------------
 // Struct type
 
+StructType::StructType(const std::vector<std::shared_ptr<Field>>& fields)
+  : NestedType(Type::STRUCT)
+{
+  children_ = fields;
```

Comment:
not a lot of code but would it make sense to refactor the population into a utility method?